design/fips-compliance.md

```diff
@@ -0,0 +1,256 @@
+# FIPS compliance for conjur-authn-k8s-client
+
+## Background
+
+The National Institute of Standards and Technology (NIST) issued the [FIPS
+ 140-2](https://csrc.nist.gov/publications/detail/fips/140/2/final) 
+Publication Series to coordinate the requirements and standards for cryptography 
+modules that include both hardware and software components.
+
+By making the conjur-authn-k8s-client FIPS compliant, our customer will gain a 
+cryptographic standard to protect their unclassified, but sensitive data.
+
+## Useful Links
+
+| **Name**                        | **Link**                                            |
+|---------------------------------|-----------------------------------------------------|
+| Epic - FIPS compliance for conjur-authn-k8s-client | https://github.com/cyberark/conjur-authn-k8s-client/issues/91 |
+| FIPS compliant crypto in golang | https://kupczynski.info/2019/12/15/fips-golang.html |
+
+## Solution
+
+We looked into 3 options for making our product FIPS compliant:
+
+1. Use cgo to call out to an existing certified library
+   
+    - This alternative disqualified due to its complexity and consuming time
+     and effort as we will have to replace any usage of crypto.
+     
+1. Use [RHEL go toolchain](https://developers.redhat.com/blog/2019/06/24/go-and-fips-140-2-on-red-hat-enterprise-linux/).
+   - RHEL takes the ownership on the encryption, and bridges between the
+    Golang encryption and OpenSSL.
+   - What needs to be done is build the Golang project on subscribed RHEL
+    machine/UBI container with the relevant go-toolset installed on it. Then, 
+    we can take the compiled binary and copy it into the containers as usual.
+   - A big advantage of this approach is that we are aligning with Red Hat
```

Comment:
It should be noted that this project isn't only available for use for big enterprise players - it's also a key part of our open source suite, and open source users may very well not care about FIPS and may not have purchased access to RedHat's tooling.